CHANGELOG.md

```diff
@@ -19,6 +19,8 @@ All versions prior to 0.9.0 are untracked.
 
 * Replaced ambient credential detection logic with the `id` package
   ([#535](https://github.com/sigstore/sigstore-python/pull/535))
+* Revamped error diagnostics reporting. All errors with diagnostics now implement
+  `sigstore._errors.Error`.
```

Comment:
The `CHANGELOG` reflects public changes, so maybe it makes sense to make `sigstore._errors` a public module? That or re-export `sigstore._error.Error` as `sigstore.Error`.

Reply:
`python-tuf` does this and it makes sense IMO, but we'd have to shuffle some things around to avoid import cycles.

Reply:
> Would we like to move all errors we expose to API consumers to `sigstore._errors`?

If we do this, we have to make sure that their existing modules re-export them, as part of not breaking out API guarantees. 

Given that, let's limit this PR to just making `sigstore._errors` public as-is, and then open a follow-up issue for consolidating our error types.